### DISCUSSION TITLE
JSON logging format.

### DISCUSSION THREAD

**Discussion by foal** [kubeapps/kubeapps#4087] [Q&A] [answered]
Hi,

Is it possible to switch the logging format of Kubeapps to JSON?  

Best,
Stas.

**Comment by antgamdia** [comment, marked as answer]
Hi @foal, I don't think there is, as of today, a direct way to enable structured JSON logs as the default output in Kubeapps. We are using `klog` as the logging library (but not in every file, though, see #3848). I guess a good starting point is the migration guide the k8s folks wrote: https://github.com/kubernetes/community/blob/master/contributors/devel/sig-instrumentation/migration-to-structured-logging.md
Alternatively, you can use a solution like `Logstash` for parsing the log entries.

If you suggest any specific approach for implementing structured logs, please feel free to send a PR!

**Reply by absoludity** [reply]
We can (and should) update in the kubeapps apisserver to using `klog.InfoS` and `klog.ErrorS` which are drop-in replacements, then users can choose to add the `--logging-format` option to the deployment (or better, we enable from the chart). At least that would have us enabling structured logging in the main service into the future.

**Reply by antgamdia** [reply]
For the record, we have standardized the logging libraries we were using (#3848 ) and we've created an issue for working on the structured log as Michael suggested: https://github.com/kubeapps/kubeapps/issues/4193

If you want to start migrating those log traces more important to your use case, please feel free to contribute :)
